packages/sdk/src/sync-agent/members/models/metadata/displayName.ts

```diff
@@ -0,0 +1,107 @@
+import { check } from '@river-build/dlog'
+import { LoadPriority, type Identifiable, type Store } from '../../../../store/store'
+import {
+    PersistedObservable,
+    persistedObservable,
+} from '../../../../observable/persistedObservable'
+import type { RiverConnection } from '../../../river-connection/riverConnection'
+import { isDefined } from '../../../../check'
+import { make_MemberPayload_DisplayName } from '../../../../types'
+
+export interface MemberDisplayNameModel extends Identifiable {
+    id: string
+    streamId: string
+    initialized: boolean
+    displayName?: string
+    isEncrypted?: boolean
+}
+
+@persistedObservable({ tableName: 'member_displayName' })
+export class MemberDisplayName extends PersistedObservable<MemberDisplayNameModel> {
+    constructor(
+        userId: string,
+        streamId: string,
+        private riverConnection: RiverConnection,
+        store: Store,
+    ) {
+        super(
+            {
+                id: `${userId}_${streamId}`,
+                streamId,
+                initialized: false,
+                displayName: undefined,
+                isEncrypted: false,
+            },
+            store,
+            LoadPriority.high,
+        )
+    }
+
+    protected override async onLoaded() {
+        this.riverConnection.registerView((client) => {
+            if (
+                client.streams.has(this.data.id) &&
+                client.streams.get(this.data.id)?.view.isInitialized
+            ) {
+                this.onStreamInitialized(this.data.id)
+            }
+            client.on('streamInitialized', this.onStreamInitialized)
+            client.on('streamDisplayNameUpdated', this.onStreamDisplayNameUpdated)
+            return () => {
+                client.off('streamInitialized', this.onStreamInitialized)
+                client.off('streamDisplayNameUpdated', this.onStreamDisplayNameUpdated)
+            }
+        })
+    }
+
+    async setDisplayName(displayName: string) {
```

Comment:
One last question - for all of these, what happens if you call them before `this.data.initialized` is true? We need to handle this case.

Reply:
```ts
test('update username', async () => {
    const userIds = space.members.data.userIds
    const metadata = space.members.getMember(bob.userId)
    const promise = metadata?.setUsername('bob123')
    expect(metadata).toBeDefined()

    await bob.start()
    await promise
    expect(userIds).toContain(bob.userId)
    expect(metadata?.username).toBe('bob123')
    await bob.stop()
})
```
This fails: `client.setDisplayName` check if the `client.stream(streamId)` is defined. 
We would need something like `riverConnection.callStream(streamId, () => T)`, which call if stream is defined or enqueue it